### ISSUE TITLE
[BUG] FuelSdk is not compatible with Java 11 

### ISSUE THREAD

**Issue by pc-dhutton** [salesforce-marketingcloud/FuelSDK-Java#117]
**Describe the bug**
When upgrading the project SDK to Java 11 FuekSdk 1.5.0 does not work. It gives the below error.

> java.lang.NoClassDefFoundError: javax/jws/WebService at org.apache.cxf.jaxws.support.JaxWsImplementorInfo.initialize(JaxWsImplementorInfo.java:273) at org.apache.cxf.jaxws.support.JaxWsImplementorInfo.<init>(JaxWsImplementorInfo.java:60) at org.apache.cxf.jaxws.JaxWsClientFactoryBean.setServiceClass(JaxWsClientFactoryBean.java:43) at org.apache.cxf.frontend.ClientProxyFactoryBean.setServiceClass(ClientProxyFactoryBean.java:224) at org.apache.cxf.jaxws.ServiceImpl.createPort(ServiceImpl.java:445) at org.apache.cxf.jaxws.ServiceImpl.getPort(ServiceImpl.java:359) at org.apache.cxf.jaxws.ServiceImpl.getPort(ServiceImpl.java:350) at javax.xml.ws.Service.getPort(Service.java:169) at com.exacttarget.fuelsdk.internal.PartnerAPI.getSoap(PartnerAPI.java:63) at com.exacttarget.fuelsdk.ETSoapConnection.<init>(ETSoapConnection.java:102) at com.exacttarget.fuelsdk.ETSoapConnection.<init>(ETSoapConnection.java:210) at com.exacttarget.fuelsdk.ETClient.buildClients(ETClient.java:219) at com.exacttarget.fuelsdk.ETClient.<init>(ETClient.java:197)

**To Reproduce**
- Change project Java SDK version from 8 to 11
- Try to create a client in a test

    		ETConfiguration config = new ETConfiguration();
			config.set("clientId", clientId);
			config.set("clientSecret", clientSecret);
			config.set("authEndpoint", authEndpoint);
			config.set("endpoint", endpoint);
			config.set("soapEndpoint", soapEndpoint);
			config.set("useOAuth2Authentication", useOAuth2Authentication);
			config.set("applicationType", applicationType);
			fuelSdkEtclient = new ETClient(config);

- Error is thrown

    `java.lang.NoClassDefFoundError: javax/jws/WebService`

**Expected behavior**
`new EtClient(config)` should initialize with the log message below

    2020-05-20 16:05:06 INFO  ReflectionServiceFactoryBean:388 - Creating Service {http://exacttarget.com/wsdl/partnerAPI}PartnerAPI from WSDL: jar:file:/Users/userName/.m2/repository/com/github/salesforce-marketingcloud/fuelsdk/1.5.0/fuelsdk-1.5.0.jar!/etframework.wsdl

**Screenshots**
![image](https://user-images.githubusercontent.com/45829775/82582686-44f34b80-9b47-11ea-9dd1-585fada1657f.png)

**Code snippet**

    		ETConfiguration config = new ETConfiguration();
			config.set("clientId", clientId);
			config.set("clientSecret", clientSecret);
			config.set("authEndpoint", authEndpoint);
			config.set("endpoint", endpoint);
			config.set("soapEndpoint", soapEndpoint);
			config.set("useOAuth2Authentication", useOAuth2Authentication);
			config.set("applicationType", applicationType);
			fuelSdkEtclient = new ETClient(config);

**Environment**
- SDK Version 11
- Java/ JDK version 11

**The bug has the severity**
- [x] Critical: The defect affects critical functionality or critical data. It does not have a workaround.
- [ ] Major: The defect affects major functionality or major data. It has a workaround but is not obvious and is difficult.
- [ ] Minor: The defect affects minor functionality or non-critical data. It has an easy workaround.
- [ ] Trivial: The defect does not affect functionality or data. It does not even need a workaround. It does not impact productivity or efficiency. It is merely an inconvenience.

**Additional context**
This solution does not work https://stackoverflow.com/questions/53727322/migration-from-jdk-8-to-jdk-11-cxf-generated-sources-unknown-imports

**Comment by lawrence-leung-mox** [comment]
A coworker ran into the same issue just because he was on jdk11. Basically the javax.* that’s missing has been removed since java 9.  

You could try importing separate dependency suggested here https://stackoverflow.com/a/54518582

Or we ended up reverting to java 8 and in the end skipping the use of this sdk all together.

Good luck

**Comment by pc-dhutton** [comment]
@lawrence-leung-mox Thank you for the reply. I tried adding the below library as stack overflow suggested and got a new error. I looked into that one for a bit and haven't found a clear solution.

    <dependency>  
       <groupId>com.sun.xml.ws</groupId>  
       <artifactId>jaxws-ri</artifactId>
       <version>2.3.0</version>  
       <type>pom</type>
    </dependency>
    
Error Message
> java.lang.ClassCastException: class com.sun.xml.ws.client.sei.SEIStub cannot be cast to class org.apache.cxf.frontend.ClientProxy (com.sun.xml.ws.client.sei.SEIStub and org.apache.cxf.frontend.ClientProxy are in unnamed module of loader 'app')
	at org.apache.cxf.frontend.ClientProxy.getClient(ClientProxy.java:128)
	at com.exacttarget.fuelsdk.ETSoapConnection.<init>(ETSoapConnection.java:103)
	at com.exacttarget.fuelsdk.ETSoapConnection.<init>(ETSoapConnection.java:210)
	at com.exacttarget.fuelsdk.ETClient.buildClients(ETClient.java:219)
	at com.exacttarget.fuelsdk.ETClient.<init>(ETClient.java:197)
	at com.personalcapital.exacttarget.client.delegate.ExactTargetFuelSdkDelegate.getFueldSdkEtclient(ExactTargetFuelSdkDelegate.java:1667)
	at com.personalcapital.exacttarget.client.delegate.ExactTargetFuelSdkDelegate.updateSubscriber(ExactTargetFuelSdkDelegate.java:1067)
	at com.personalcapital.exacttarget.service.ExactTargetServiceImpl.unsubscribeUserByUserGuid(ExactTargetServiceImpl.java:155)
	at com.personalcapital.exacttarget.client.delegate.ExactTargetFuelSdkDelegateTest.fuelSdkUnsubscribeSubscriber(ExactTargetFuelSdkDelegateTest.java:136)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.springframework.test.context.junit4.statements.RunBeforeTestExecutionCallbacks.evaluate(RunBeforeTestExecutionCallbacks.java:74)
	at org.springframework.test.context.junit4.statements.RunAfterTestExecutionCallbacks.evaluate(RunAfterTestExecutionCallbacks.java:84)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:251)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunnerRepeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)

**Comment by lawrence-leung-mox** [comment]
@pc-dhutton trying to get it working with java 11 will probably take more time than considering falling back to using a HTTP Client and connecting to SFMC thru REST API, or perhaps even downgrading your whole project to java 8.  I know they are not ideal but are viable alternatives.

**Comment by pc-dhutton** [comment]
@lawrence-leung-mox same I agree. I'll leave the ticket open and maybe 
Salesforce Marketing Cloud will do something about it. Leaning towards a separate microservice that runs Java 8 while the rest of the stack runs Java 11

**Comment by waldou** [comment]
Works on Java 11 using this dependencies:

	implementation 'com.github.salesforce-marketingcloud:fuelsdk:1.5.0'
	implementation 'jakarta.xml.ws:jakarta.xml.ws-api:2.3.3'
	implementation 'com.sun.xml.messaging.saaj:saaj-impl:1.5.1'

**Comment by pgiedrojc** [comment]
For me it worked on Java 11 with those dependencies.
Funny part that if you put higher versions it simply won't work

`<dependency>
        <groupId>javax.xml.bind</groupId>
        <artifactId>jaxb-api</artifactId>
        <version>2.3.0</version>
    </dependency>
    <dependency>
        <groupId>com.sun.xml.bind</groupId>
        <artifactId>jaxb-core</artifactId>
        <version>2.3.0</version>
    </dependency>
    <dependency>
        <groupId>com.sun.xml.bind</groupId>
        <artifactId>jaxb-impl</artifactId>
        <version>2.3.0</version>
    </dependency>
    <dependency>
        <groupId>com.sun.xml.messaging.saaj</groupId>
        <artifactId>saaj-impl</artifactId>
        <version>1.5.0</version>
    </dependency>
    <dependency>
        <groupId>javax.xml.ws</groupId>
        <artifactId>jaxws-api</artifactId>
        <version>2.2.6</version>
    </dependency>`

Hope it helps

**Comment by pc-dhutton** [comment]
Ok sweet! Thank you for the update. I'll try that within the next few days and report back

**Comment by red-avtovo** [comment]
@pc-dhutton you can take a look on my PR. It successfully runs on Java 11 now

**Comment by red-avtovo** [comment]
Hey everyone!
Since #125 is merged this ticket is finally resolved. Thx to some external force which help us to move things forward.
Congrats everyone for being able to finally use the library on Java 11
The new version 1.5.1 was also [published](https://search.maven.org/artifact/com.github.salesforce-marketingcloud/fuelsdk/1.5.1/jar) 🎉

**Comment by pc-dhutton** [comment]
Confirmed that @red-avtovo fix has worked in 1.5.1! Thank you very much sir. Also, I found that our monolithic app was using older versions of saaj and the required SOAP dependencies for Fuel to work. These older dependencies created conflicts with Fuel which made it incompatible with the mono. So I separated Fuel 1.5.1 out into its own spring boot microservice running java 11 and it works perfectly now.
